### ISSUE TITLE
Working with workspace-inheritance

### ISSUE THREAD

**Issue by aesteve** [LukeMathWalker/cargo-chef#148]
Hello, and thanks for your work on `cargo-chef`

I tried experimenting with the new (unstable, 1.64) feature of [workspace-inheritance](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#workspace-inheritance) and faced the following issue.

With 2 members in the workspace, inheriting some common dependencies, and defining their own. (very close to the doc above)

And got the following error (`docker build`): 
```
 > [planner 2/2] RUN cargo +nightly chef prepare --recipe-path recipe.json:
#14 0.270 Error: Failed to compute recipe
#14 0.270
#14 0.270 Caused by:
#14 0.270     0: invalid type: map, expected a string for key `package.version` at line 3 column 21
#14 0.270     1: invalid type: map, expected a string for key `package.version` at line 3 column 21
```

`package.version` being defined as in the docs as: `version.workspace = true`.

The Dockerfile I'm using to build is [here](https://github.com/aesteve/buska/blob/457e5d6ce068b73025f1eb578e8f55d7233bcf75/Dockerfile) and the overall WIP I'm working on is [there](https://github.com/aesteve/buska/pull/5/files).

Let me know if I can do something to help you

**Comment by LukeMathWalker** [comment]
Hey!
This won't work because `cargo-manifest` (used by `cargo-chef`) doesn't know how to parse the manifests using the new unstable feature.

I'm afraid I am not open to PRs on this until the feature stabilises and lands in `beta` - I don't want to keep up with nightly breakage.

**Comment by lasantosr** [comment]
@LukeMathWalker The feature is stabilized and it's already on current beta (1.64)

https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#workspace-inheritance-1

**Comment by LukeMathWalker** [comment]
If you run `cargo-upgrade` to pull in the latest version of `cargo-manifest` it _should_ work 👍🏻

**Comment by aesteve** [comment]
Nice! Thank you.
Any idea where I should include the `cargo-upgrade` here: 

```Dockerfile
FROM rustlang/rust:nightly-bullseye-slim as chef
RUN cargo +nightly install cargo-chef --locked
WORKDIR app

FROM chef AS planner
COPY . .
RUN cargo +nightly chef prepare --recipe-path recipe.json

FROM chef AS builder

COPY --from=planner /app/recipe.json recipe.json
# Build dependencies - this is the caching Docker layer!
RUN apt update
RUN apt install -y cmake
RUN apt install -y libssl-dev
RUN cargo +nightly chef cook --release --recipe-path recipe.json
# Build application
COPY . .
RUN cargo +nightly build --all --release

FROM debian:bullseye-slim AS runtime
WORKDIR app
RUN apt update
RUN apt install -y libssl1.1 openssl ca-certificates
COPY --from=builder /app/target/release/buska-cli /usr/local/bin

ENTRYPOINT ["/usr/local/bin/buska-cli"]
```

Or maybe I should just get rid of some cached docker layer?
'Cause I gave it a try and got the same error, but that must be because I didn't catch the latest `cargo-manifest`  version.

**Comment by LukeMathWalker** [comment]
You need to run it on the host machine to upgrade the contents of your `Cargo.toml`.
I will also cut a release to upgrade the pre-built Docker image once it's confirmed to be working.

**Comment by aesteve** [comment]
No problem at all. There's no urgency, I just wanted to give this new feature a try and figured out it could be interesting for you to know cargo-chef wasn't compatible, yet.

Tell me if you want me to give it a try, or if I can help you with anything.

Thanks again for your work on `cargo-chef`!!

**Comment by LukeMathWalker** [comment]
Done - the change should have been released with `0.1.40` 😁

**Comment by lasantosr** [comment]
I can confirm it is working now as expected! Thanks 😀

I've only tried this style: `regex = { workspace = true, features = ["unicode"] }`, so I can't confirm about `cc.workspace = true`.

**Comment by aesteve** [comment]
I'm sorry I must still be missing something: 

```
❯ cargo +nightly chef --version
cargo-chef 0.1.40
```
=>
```
❯ cargo +nightly chef prepare --recipe-path /tmp/recipe.json
Error: Failed to compute recipe

Caused by:
    0: invalid type: map, expected a string for key `package.version` at line 3 column 21
    1: invalid type: map, expected a string for key `package.version` at line 3 column 21
```

**Comment by LukeMathWalker** [comment]
Can you share your `Cargo.toml`? @aesteve

**Comment by aesteve** [comment]
Everything is on this branch: https://github.com/aesteve/buska/blob/feat/upgrades/Cargo.toml

**Comment by LukeMathWalker** [comment]
Ok, the issue is related to this line: https://github.com/aesteve/buska/blob/33414b1c1c0ea15d2f42ac29476129352af0edf4/cli/Cargo.toml#L3

It will take a bit more work to sort this one out I think, based on how `cargo-chef` works.

**Comment by Voxelot** [comment]
Also filed an issue here: https://gitlab.com/crates.rs/cargo_toml/-/issues/18

**Comment by LukeMathWalker** [comment]
I have implemented the missing pieces in `0.1.43`/`0.1.44`. I can confirm that your branch builds correctly @aesteve 👌🏻 
We should have everything needed to support the inheritance features stabilised in Rust 1.64.0 🎉
